### ISSUE TITLE
Add WebP support

### ISSUE THREAD

**Issue by anijatsu** [ShareX/ShareX#765]
As in the title. It's more efficient than JPEG and whatever else, and it'd be pretty cool to be the first screenshot app with this format's support.

**Comment by Jaex** [comment]
When none of the browsers support it what is point.

**Comment by anijatsu** [comment]
It's getting there, Chromium has got it working, that means default Android browsers since 4.1, Opera & Opera Mini, Chrome 31 and up. http://caniuse.com/#feat=webp
Would there be a possibility to make it as a plugin for ShareX?

**Comment by Jaex** [comment]
And what about Firefox or other browser users they won't see anything? Is it acceptable for them to not see anything?

**Comment by anijatsu** [comment]
Then why there's BMP or TIFF in the options? These formats aren't too widely acceptable for online storage.

**Comment by Jaex** [comment]
Both bmp and tiff openable by default in Windows. WebP not.
Also adding bmp and tiff support is like 5 seconds job.

**Comment by RedMser** [comment]
@Jaex @anijatsu I started working on **[WebP support](https://github.com/RedMser/ShareX/tree/webpsupport) for ShareX**, using Jose M. Piñeiro's [light-weight WebP wrapper](https://github.com/JosePineiro/WebP-wrapper) (GPL v3).

Right now, you can select it as an image format for capturing WebP files that can be saved to file or uploaded to an image host that accepts WebP. The "JPEG quality" setting is reused for lossy encoding quality right now. Lossless encoding is *supported*, but no setting exists for it yet.

![gimp-2 9_2018-02-04_13-38-38](https://user-images.githubusercontent.com/5117197/35777584-c5f6b418-09b0-11e8-8b31-edd15960f8e4.png)

Main issue right now is that there are multiple methods for loading and saving image files/streams in ShareX's code base, and each needs different handling for WebP files (so for example, previews don't work and the different tools that allow loading and saving images don't support the format).

But it definitely is possible and could be fully integrated into the program with a bit more work! :)

**Comment by Jaex** [comment]
I'm trying to decrease external dependencies in ShareX not add more.

**Comment by xnaas** [comment]
With the ability to record animated WebPs, will we see the option to take static WebP screenshots as well?

**Comment by CommenterOfComments** [comment]
I _really_ static WebP support. At this point every major browser supports WebP, except for IE (which is dead) and Safari (which apple is... _courageous_). I'm attempting to switch to WebP as much as I can for images, and ShareX is somewhat limiting that. I think at this point it's worth the effort, considering the now wide support and large benefits of the format. I'd also love to see additional options for stuff like alpha compression, near lossless, and compression time like with [cwebp](https://developers.google.com/speed/webp/docs/cwebp).

**Comment by Jaex** [comment]
Technically you can still use WebP output using actions feature and CLI app which can convert lossless PNG to WebP.

**Comment by CommenterOfComments** [comment]
That's a pretty good fix for now, and I'll probably use it, but I still think there should be official support. Command line tools aren't really user-friendly after all, most people probably wouldn't even try.

**Comment by FozzieHi** [comment]
@Jaex So will this be added? not sure why it was closed as support was evidently going to increase over time. Now Safari also supports WebP so the only incompatible browser is IE.

**Comment by axefrog** [comment]
@Jaex All modern browsers support WebP now. Even large image sharing sites such as gfycat now use it everywhere. There's almost no reason to use animated gifs anymore.

**Comment by ledlamp** [comment]
BUMP! We're reaching the point where all browsers support WebP now; Safari is the only browser that doesn't but it will very soon with version 14. Discord also fully supports embedding WebP images.

I have slow upload bandwidth (3mbps) so PNG screenshots of games and photographic stuff like that take a long time to upload. JPEG uploads _much_ faster but it doesn't support transparency which I need for the shadow effect applied to my screenshot. As I mainly post them on Discord I set the background color to match discord to work-around it but this isn't ideal. It would be very useful to have a format that supports both lossy compression _and_ transparency. While also being even more efficient with size, helping images upload even faster. And WebP may also be a significant upgrade over GIFs, which are terribly efficient with file size, although platforms like discord don't support them yet.

I really think it's time to implement WebP support.

**Comment by SimplyCorbett** [comment]
Will you be adding WebP support?

**Comment by sergeevabc** [comment]
ShareX on my end believes WebP are generic files, not images, so it does not use custom image uploader. How do I fix that?

**Comment by Jaex** [comment]
> ShareX on my end believes WebP are generic files, not images, so it does not use custom image uploader. How do I fix that?

Add webp manually to here:

![](https://jaex.getsharex.com/2021/05/explorer_hwUDgx1wGc.png)

**Comment by Spacellary** [comment]
Here's to hoping that 2023 is the year of official webp screenshots support in ShareX

**Comment by bolach** [comment]
> Here's to hoping that 2023 is the year of official webp screenshots support in ShareX

I hope so. Google now gives priority to .webp images

**Comment by Gorvaine** [comment]
What's the best way to enable webp screenshots?

**Comment by Knewest** [comment]
WebP is the future. JPEG, PNG and GIF need to all move aside for this faster, newer, more efficient codec/format. 
It's starting with AV1 (successor to VP9 (video codec based of WebP)), now we continue with adding WebP support to screenshots, I have not found a single program that can do it. Be the first! 

Don't let companies like Adobe and Apple hinder the success, Adobe refused to implement it till 2022 and then again refused to _properly_ implement it to this day, despite implementing Apple's codecs down to a T. Apple refused to add WebP support till 2020 and even removed MPEG-2 support in favour of their own codec-- Why not both?  Because of anti-consumerism. Adobe and Apple want to be _the_ company to release the codec that replaces the three giants of formats. Well it's too late! Google has been developing this for over a decade and it is amazing, it needs to become the new standard. 

One day we won't have photographers who spend thousands of dollars on equipment and software only to master them in the volatile JPEG format..

Format slander and anti-consumerism is WebP's biggest enemy. 
Even companies such as Discord have begun the full transition to WebP.

I hope to see the web and all future software become faster with the implementation of WebP, all without destroying the content displayed. 

TL;DR: Just add WebP support, ShareX. I mean seriously, there is enough support for it that your "What is the point?" argument falls flat on it's face. Paint.net is still the best program for encoding WebP files.

**Comment by cvladan** [comment]
First of all, I'd like to thank the author for this amazing tool and for the time he's put into its development.

I also fully agree and understand the author's desire to reduce external dependencies in ShareX, but I believe it would be better to implement this reduction in features that are quite surprising to be part of the tool at all, and which are also present in competing tools from much more powerful creators (PowerToys from MS). I'm talking about OCR, DNS Changer, Hash Check, Borderless Window, Colour Picker and similar tools that don't really belong in a screenshot tool.

However, image format support IS a fundamental part of such a tool.

After all, ShareX didn't achieve its current success by ignoring user requests for four years and not wanting to implement the current de facto web standard for images, which is WebP.

Seriously, mate, I think you're a genius! But a pretty stubborn genius 😉

**Comment by cvladan** [comment]
Sorry, 8 years 😀

**Comment by isXander** [comment]
Animated WEBP is something I'd really like in sharex. GIF format is extremely inefficient compared to WEBP so storage would be a lot nicer using webp.

**Comment by Knewest** [comment]
> Animated WEBP is something I'd really like in sharex. GIF format is extremely inefficient compared to WEBP so storage would be a lot nicer using webp.

Hey, this is actually already added (lossless option seems to be broken, though). It is just static WebP that still has not been added.

<a href="https://cdn.discordapp.com/attachments/753561208073879642/1112450897549738064/yZ6Rc9aQ6b.webp">
  <img src="https://cdn.discordapp.com/attachments/753561208073879642/1112450897549738064/yZ6Rc9aQ6b.webp" alt="How to do it." style="cursor:pointer;"/>
</a>

**Comment by isXander** [comment]
> > Animated WEBP is something I'd really like in sharex. GIF format is extremely inefficient compared to WEBP so storage would be a lot nicer using webp.
> 
> Hey, this is actually already added (lossless option seems to be broken, though). It is just static WebP that still has not been added.
> 
> [ ![How to do it (File too large to embed, must click to view).](https://camo.githubusercontent.com/0afdd45314fc2544f33d14a7285b7158e0ec5f16c3a16d6d264ff4d6118d4c6c/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3735333536313230383037333837393634322f313131323435333836343535303035353939362f71304b5a6934313631302e77656270) ](https://cdn.discordapp.com/attachments/753561208073879642/1112453864550055996/q0KZi41610.webp)

Means you need to choose between MP4 and webP, and rather have the options to choose between the two with gif vs regular recording

**Comment by AlexeyAdamitsky** [comment]
This is something I'd really like to see supported. WebP proved to be superior to JPG in my personal experience. Size is lighter or equal but the quality is much better.

**Comment by Knewest** [comment]
Hi, all,
WebP binary executable encoder/decoder has been update to v1.3.1. For those using the actions menu for a makeshift WebP screenshot tool, you should update! I have found that it is faster than v1.3.0.<br>
Link to v1.3.1: [https://developers.google.com/speed/webp/docs/precompiled](https://developers.google.com/speed/webp/docs/precompiled)

**Comment by TariqMK** [comment]
Adding my vote for this, WebP is everywhere now ... everywhere except ShareX!

**Comment by bolach** [comment]
> Adding my vote for this, WebP is everywhere now ... everywhere except ShareX!

This will be available in the next year, for the special 10 years anniversary of this request!
No, it is not. I'm joking.

**Comment by Dioarya** [comment]
2024 for the year of WebP support on ShareX!

**Comment by Knewest** [comment]
> 2024 for the year of WebP support on ShareX!

I doubt it, honestly. 😕 

I heard that when JPEG XL becomes more adopted, WebP will finally be added! 😆

**Comment by bolach** [comment]
Probably half of the users on this request are already retired and we still don't have webp support :)

**Comment by cptvincer** [comment]
Another bump. If not webp maybe some of the other formats that squeeze less size for similar results.

In my use im constantly taking screenshots, sometimes in bulk, dozens or more each day of the entire screen (wich is 4k)... they add up quickly. I do manual conversions later but it ends up eating a lot of my time...

Chances are if/when i find some alternative that can screencap in webp i will leave sharex in a heartbeat, despite all my love for it and for FOSS as a whole.

**Comment by ellioteb** [comment]
Yep, Webp support would be great.

**Comment by kleineluka** [comment]
WebP is the standard now.

**Comment by ni6hant** [comment]
It is time my friend.

**Comment by Gorvaine** [comment]
10 years and still waiting for basic webp encoding.

**Comment by ni6hant** [comment]
With a bit of setup and following [this blog](https://limour.xlog.app/WEBP-jie-tu-gong-ju-ShareX--imagemagick?locale=en) I was able to do it and now all my images are automatically saved as webp.

P.S: If you want a video, react to this comment but the blog is pretty straight forward. I think there was one quirk but I forgot what it was.

P.P.S: It's an open source project so I completely understand that this might not happen. If I ever start making more than livable income I will surely donate.

<img width="1787" height="607" alt="Image" src="https://github.com/user-attachments/assets/285c76e5-cf5c-4fe9-bc10-8f478849e3d6" />

**Comment by Jaex** [comment]
@ni6hant I was also explained how to do it at here: https://getsharex.com/actions

**Comment by WirlyWirly** [comment]
Converting images to .webp is great and all, but being able to preview them from ShareX is still not possible.

**Comment by God-damnit-all** [comment]
> With a bit of setup and following [this blog](https://limour.xlog.app/WEBP-jie-tu-gong-ju-ShareX--imagemagick?locale=en) I was able to do it and now all my images are automatically saved as webp.

Well, whatever that site was is gone now. Not on Wayback Machine either.

**Comment by God-damnit-all** [comment]
Trying to see the progress of https://github.com/ShareX/ShareX/pull/8151 has me suddenly realizing that pull requests are closed. When did that happen?

**Comment by flawrist** [comment]
After years of reasoning the stubborn developer to adopt WebP, keep legacy [OCR](https://github.com/ShareX/ShareX/issues/6350), or fix [Beautifier](https://github.com/ShareX/ShareX/issues/7969) window, I’d like to take this opportunity to say that ShareX has turned into a bloated, sluggish mess. Version 17.0.1 was the last gasp of anything remotely usable, move on.

**Comment by BrycensRanch** [comment]
> After years of reasoning the stubborn developer to adopt WebP, keep legacy [OCR](https://github.com/ShareX/ShareX/issues/6350), or fix [Beautifier](https://github.com/ShareX/ShareX/issues/7969) window, I’d like to take this opportunity to say that ShareX has turned into a bloated, sluggish mess. Version 17.0.1 was the last gasp of anything remotely usable, move on.

This is not an airport; you do not need to announce your departure. Webp support is coming.

**Comment by navjotjsingh** [comment]
When?

**Comment by navjotjsingh** [comment]
Why have you deleted all the pull requests @Jaex That's not how open-source is supposed to work.

**Comment by BrycensRanch** [comment]
> Why have you deleted all the pull requests [@Jaex](https://github.com/Jaex) That's not how open-source is supposed to work.

As with open source, there are no ETAs.

Here is an article to understand what maintainers are dealing with: https://stack72.dev/the-community-pull-request-is-dead/#what-maintainers-are-actually-dealing-with

**Comment by God-damnit-all** [comment]
> After years of reasoning the stubborn developer to adopt WebP, keep legacy [OCR](https://github.com/ShareX/ShareX/issues/6350), or fix [Beautifier](https://github.com/ShareX/ShareX/issues/7969) window, I’d like to take this opportunity to say that ShareX has turned into a bloated, sluggish mess. Version 17.0.1 was the last gasp of anything remotely usable, move on.

I think the new image editor is nice but I'm definitely not a fan of Avalonia's whole aesthetic. It feels very cheap and new age techbro.

More importantly, the lack of transparency is concerning. I couldn't find a single mention of WHY pull requests are closed. I did find chat messages of the developer making fun of someone who wanted to submit a PR though, which seems in bad taste when not a single explanation has been provided anywhere.

**Comment by navjotjsingh** [comment]
Its time to look for Sharex alternative on Windows.

**Comment by navjotjsingh** [comment]
Also, I think he should delete the Issues as well. What's the point when we are not valued?

**Comment by God-damnit-all** [comment]
> Its time to look for Sharex alternative on Windows.

> Also, I think he should delete the Issues as well. What's the point when we are not valued?

That's a bit much. Some of the recent activity has been worrying but let's not be too hasty.

**Comment by Polda18** [comment]
I'm already eyeing an alternative for ShareX, which is actually its fork SnapX. But not for the reason that ShareX might be bloated. I'm quite happy with it, but the only downside I experience with it is no Linux support. As I'm slowly shifting more towards Linux and slowly abandoning Windows (not entirely, but enough to consider running Linux alongside Windows in a dualboot setup), I will absolutely need a crossplatform solution. Since @Jaex isn't familiar with the Linux environment and thus isn't particularily interested in introducing a Linux support, an alternative is a necessity for me. Not really a choice. I would stay with the original ShareX, if it provided a cross-platform solution natively.

**Comment by God-damnit-all** [comment]
> As I'm slowly shifting more towards Linux and slowly abandoning Windows (not entirely, but enough to consider running Linux alongside Windows in a dualboot setup)

For what it's worth, if this is due to the whole Windows 11 debacle, you should know that because of just how hard the OS has been flopping they started up something called Project K2 that the veteran developers of the company have been leading, rather than Satya Nadella's army of web UI techbros. 26H2, which is likely coming this fall, will showcase some of those efforts.

27H2 next year is going to be the real splash though, where Windows 11 actually earns its right to exist by being the first OS to standardize a dual-architecture platform (x64 & ARM64) without forcing developers to completely rewrite or dual-compile their existing apps.

> Since [@Jaex](https://github.com/Jaex) isn't familiar with the Linux environment and thus isn't particularily interested in introducing a Linux support, an alternative is a necessity for me.

Well, Linux Desktop software has long been notoriously difficult to support, and it seems like the project is having a hard enough time just supporting Window. Wayland has also really made a mess of things so now the problem is even worse. Adapting a project like this to all platforms would be very difficult.

Have you looked into the successor project that they're working on, XerahS? Cross-platform support is part of the core design. https://xerahs.com

**Comment by Knewest** [comment]
> Its time to look for Sharex alternative on Windows.

I've felt this way for a long time with how dated the UI looks and the developer's total disinterest to even comment on anything. Even back when I first started posting in the WebP issue threads back in 2023, it was really taboo to even discuss WebP in any positive way within the ShareX Discord server (regardless of how objectively factual what you were saying was).

In a way I'm not surprised how this turned out. Old habits die old, and this issue is _ooollldddd_. So old that I'm now pretty capable of creating my own pull request, but am I really going to waste my time with another developer I know doesn't look at technical advancments objectively?

**Comment by God-damnit-all** [comment]
> I've felt this way for a long time with how dated the UI looks

Well, good news for you then, the UI that is currently in ShareX's beta has a much more modern look, similar to the screenshots of this related project by the same team: https://xerahs.com

Personally, I hate it and have never had a problem with the way ShareX looked beforehand (especially after dark mode was added).

> So old that I'm now pretty capable of creating my own pull request, but am I really going to waste my time with another developer I know doesn't look at technical advancments objectively?

Actually, you're not. This repo closed pull requests to the public a little over a month ago, it seems.
